ods_ci/tasks/Resources/RHODS_OLM/install/oc_install.robot

```diff
@@ -45,31 +42,22 @@ Install RHODS
            IF  "${UPDATE_CHANNEL}" != "odh-nightlies"
                 Install RHODS In Managed Cluster Using CLI  ${cluster_type}     ${image_url}
            ELSE
-                ${operator_csv_name} =    Set Variable    ${ODH_OPERATOR_NAME}
                 Create Catalog Source For Operator
                 Oc Apply    kind=List    src=tasks/Resources/Files/odh_nightly_sub.yml
            END
       ELSE
           FAIL    Provided test envrioment is not supported
       END
   END
-  Wait Until Csv Is Ready    ${operator_csv_name}
+  Wait Until All Csv Are Ready
```

Comment:
It is missleading, since ${operator_csv_name} is apparently not the full name of the CSV, but only the prefix 'rhods-operator' without the CSV version. 
Having the verification to verify all CSVs, no matter which version is safer.

If we do not do so, we will have to know the exact CSV name which is currently installed.

Why not to verify all ?

Reply:
If you prefer, I can filter by label instead:
```
$▶ oc wait --timeout=3m --for jsonpath='{.status.phase}'=Succeeded -l olm.copiedFrom=redhat-ods-operator csv -n openshift-operators
clusterserviceversion.operators.coreos.com/rhods-operator.2.6.0 condition met
```

Reply:
Agree that the name of the variable may not be the correct one, then we could rename.

My concern with this change is that with this we expect that all CSVs on the cluster are in Ready state. What if there is some CSV that isn't odh/rhoai operator tied, that isn't in the Ready state? In such situation this check fail. I don't think we want that.

Reply:
I will keep it as you suggested, just adding a side note:

A failed CSV in the same operators namespace as RHOAI, will indicate something is wrong. It is safer to verify `--all` other CSVs, which we usually install as part of `Install RHODS` Keyword, e.g:
```
clusterserviceversion.operators.coreos.com/openshift-pipelines-operator-rh.v1.12.2 condition met
clusterserviceversion.operators.coreos.com/serverless-operator.v1.31.0 condition met
clusterserviceversion.operators.coreos.com/servicemeshoperator.v2.4.5 condition met
```

If one of them failed, whether installed via odh operator bundle, or as part of our test flow in ODS-CI, 
it should raise a red flag if their CSV is not ready on time.

Reply:
I don't object against checking all of these. But let's do it on proper place. These all keywords are dealing just with the RHOAI/ODH operator. Thus it will be confusing if it suddenly fail because some other CSV isn't in a good shape.

I believe that such check should be done with each operator installation. So it is taken care per operator basis. If we want to do a global check for all of these on one place, then it has to be a correct place so it is obvious also from used keywords...

Reply:
Doing it for each operator CSV will require to identify the specific label for the CSV, which is quite unstructured, e.g:
```
NAME                   DISPLAY                VERSION   REPLACES               PHASE       LABELS
rhods-operator.2.6.0   Red Hat OpenShift AI   2.6.0     rhods-operator.2.5.0   Succeeded   olm.copiedFrom=redhat-ods-operator

NAME                          DISPLAY                  VERSION   REPLACES   PHASE       LABELS
opendatahub-operator.1.18.0   Open Data Hub Operator   1.18.0               Succeeded   operators.coreos.com/rhods-operator.openshift-operators=
```

Unfortunately CSV does not support field selector on display name, so we need to define these unique label variables, which may be easily changed in the future:
```
${RHODS_CSV_LABEL}=    olm.copiedFrom\=redhat-ods-operator
${ODH_CSV_LABEL}=    operators.coreos.com/rhods-operator.openshift-operators
```

Reply:
Yeah... TBH, I would prefer to stick with the original approach to compare against name prefix. I suppose that we can't use it because of the limitation of the `oc wait`, correct?

Reply:
Yes, because oc wait can only handle one condition (did status phase succeed ?) but not another one (does annotation include specific name ? ).

I changed the logic to get the CSV name first, and then wait for its status.